qt/widgets/instrumentview/src/InstrumentRendererMultiList.cpp

```diff
@@ -0,0 +1,89 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidQtWidgets/InstrumentView/InstrumentRendererMultiList.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidQtWidgets/InstrumentView/InstrumentActor.h"
+#include "MantidQtWidgets/InstrumentView/OpenGLError.h"
+
+using Mantid::Beamline::ComponentType;
+
+namespace MantidQt::MantidWidgets {
+InstrumentRendererMultiList::InstrumentRendererMultiList(const InstrumentActor &actor) : InstrumentRenderer(actor) {
+  const auto &componentInfo = actor.componentInfo();
+
+  m_useNonPickingDisplayList = false;
+  m_usePickingDisplayList = false;
+
+  m_pickingDisplayListId.resize(componentInfo.size());
+  m_nonPickingDisplayListId.resize(componentInfo.size());
+  for (size_t i = 0; i < componentInfo.size(); ++i) {
+    m_nonPickingDisplayListId[i] = 0;
+    m_pickingDisplayListId[i] = 0;
+  }
+}
+
+InstrumentRendererMultiList::~InstrumentRendererMultiList() {
+  for (unsigned int i : m_nonPickingDisplayListId) {
+    if (i != 0) {
+      glDeleteLists(i, 1);
+    }
+  }
+  for (unsigned int i : m_pickingDisplayListId) {
+    if (i != 0) {
+      glDeleteLists(i, 1);
+    }
+  }
+}
+
+void InstrumentRendererMultiList::renderInstrument(const std::vector<bool> &visibleComps, bool showGuides,
+                                                   bool picking) {
+  if (std::none_of(visibleComps.cbegin(), visibleComps.cend(), [](bool visible) { return visible; }))
+    return;
+
+  if (!instrActor().isInitialized()) {
+    return;
+  }
+
+  OpenGLError::check("InstrumentActor::draw()");
+  const auto &displayListIds = picking ? m_pickingDisplayListId : m_nonPickingDisplayListId;
+  auto &useDisplayList = picking ? m_usePickingDisplayList : m_useNonPickingDisplayList;
```

Comment:
Feels as if maybe these bools could be simplified? is it always either `m_usePickingDisplayList` or `m_useNonPickingDisplayList`? In that case could they be one bool? I might be misunderstanding.